File: 1/1.4/Python/1.4.py
def all_perms(elements):
    if len(elements) <= 1:
        yield elements
    else:
        for perm in all_perms(elements[1:]):
            for i in range(len(elements)):
                yield perm[:i] + elements[0:1] + perm[i:]


def isPalindrom(s: str) -> bool:
    for i in range(int(len(s)/2)):
        if s[i] != s[len(s) -1 -i]:
            return False
    return True


def isPalindromAnagram(s: str) -> bool:
    s = s.replace(" ", "")

    for permutation in all_perms(s):
        if isPalindrom(permutation):
            print("Palindrom found: {}".format(permutation))
            return True
    return False

if __name__ == '__main__':
    print(isPalindromAnagram("ar"))
    print(isPalindromAnagram("ara"))
    print(isPalindromAnagram("raa"))
    print(isPalindromAnagram("ar "))
    print(isPalindromAnagram("a ra"))
    print(isPalindromAnagram("r aa"))

File: 1/1.4/С++/1.4/1.4.cpp
#include <iostream>
#include <string>
#include <regex>
#include "generator.hpp"

tl::generator<std::string> all_perms(std::string elements) {
	if( elements.length() <= 1 ){
		co_yield elements;
	}
	else {
		for( auto perm : all_perms(elements.substr(1))) {
			for( int i = 0;  i < elements.length(); i++) {
				std::string res = perm.substr(0, i) + elements.substr(0, 1) + perm.substr(i);
				co_yield res;
			}
		}
	}
}

bool isPalindrome(std::string s) {
	for (int i = 0; i < s.length() / 2; i++) {
		if( s[i] != s[s.length() - 1 - i]){
			return false;
		}
	}
	return true;
}

bool isPalindromeAnagram(std::string s) {
	s = std::regex_replace(s, std::regex(" "), "");

	for(std::string permutation : all_perms(s)) {
		if( isPalindrome(permutation)) {
			std::cout << "Palindrome found: " + permutation << std::endl;
				return true;
		}
	}
	return false;
}

int main()
{
	std::cout << isPalindromeAnagram("ar") << std::endl;
	std::cout << isPalindromeAnagram("ara") << std::endl;
	std::cout << isPalindromeAnagram("raa") << std::endl;
	std::cout << isPalindromeAnagram("ar ") << std::endl;
	std::cout << isPalindromeAnagram("a ra") << std::endl;
	std::cout << isPalindromeAnagram("r aa") << std::endl;
	std::cout << isPalindromeAnagram("r ara") << std::endl;
}
